astropy/convolution/convolve.py

```diff
@@ -869,3 +887,30 @@ def convolve_models(model, kernel, mode='convolve_fft', **kwargs):
         raise ValueError(f'Mode {mode} is not supported.')
 
     return CompoundModel(mode, model, kernel)
+
+
+def _next_fast_lengths(shape):
+    """
+    Find optimal or good sizes to pad an array of ``shape`` to for better
+    performance with `numpy.fft.*fft` and `scipy.fft.*fft`.
+    Calculated directly with `scipy.fft.next_fast_len`, if available; otherwise
+    looked up from list and scaled by powers of 10, if necessary.
+    """
+
+    try:
+        import scipy.fft
+        return np.array([scipy.fft.next_fast_len(j, real=False) for j in shape])
+    except ImportError:
+        pass
+
+    newshape = np.empty(len(np.atleast_1d(shape)), dtype=int)
+    for i, j in enumerate(shape):
+        scale = 10 ** max(int(np.ceil(np.log10(j))) - _good_range, 0)
+        for n in _good_sizes:
+            if n * scale >= j:
+                newshape[i] = n * scale
```

Comment:
I have a minor preference, and I think astropy style docs (?) also have, to avoid 1-letter variables

Reply:
I thought for small loops and the like they were accepted; in fact I could not find any recommendation against them in the developer guide and there are some examples even for other use (see e.g. `utils.py`).
But I agree they may well be more descriptive.

Reply:
yeah, I won't complain for inline loops, and I'm ambivalent here (like you said, this is a small enough loop), but it's on the border and I tend to err on the side of longer variable names.